packages/server/src/integrations/rest.ts

```diff
@@ -5,8 +5,52 @@ import {
 } from "../definitions/datasource"
 import { IntegrationBase } from "./base/IntegrationBase"
 
+const BodyTypes = {
+  NONE: "none",
+  FORM_DATA: "form",
+  ENCODED: "encoded",
+  JSON: "json",
+  TEXT: "text",
+}
+
+const coreFields = {
+  path: {
+    type: DatasourceFieldTypes.STRING,
+    display: "URL",
+  },
+  queryString: {
+    type: DatasourceFieldTypes.STRING,
+  },
+  headers: {
+    type: DatasourceFieldTypes.OBJECT,
+  },
+  enabledHeaders: {
+    type: DatasourceFieldTypes.OBJECT,
+  },
+  requestBody: {
+    type: DatasourceFieldTypes.JSON,
+  },
+  bodyType: {
+    type: DatasourceFieldTypes.STRING,
+    enum: Object.values(BodyTypes),
+  },
+}
+
 module RestModule {
   const fetch = require("node-fetch")
+  const { formatBytes } = require("../utilities")
+  const { performance } = require("perf_hooks")
+
+  interface RestQuery {
```

Comment:
Just realising now that this base query I have added isn't very generic
https://github.com/Budibase/budibase/blob/develop/packages/server/src/definitions/common.ts#L121-L125

We could link these two by renaming this one you have added as `RestQueryFields` and replacing fields block in the common definition with
```js
fields: RestQueryFields
```

I'm not sure if the fields key is used by other types of query, if so we can type them and expand the definition e.g.
```js
fields: RestQueryFields |  PostgresFields | S3Fields
```

Should we move this one out into the definitions folder as well?

Reply:
yeah we probably should, it is used by basically all query types, so it might just need an `any` on the end for now as there isn't interfaces for everything, it really can contain anything.

Reply:
aye right enough, for now we could do
```js
fields: RestQueryFields | any
```

during the import stuff I came across `is` functions for typescript for safe casting where we can have a function like:
```
const isRestQuery(query: Query, fields: any) fields is RestQueryFields => {
   query.type === "REST"
} 
```
and then get typesafe usage on a multi type field by 
```
const fields = query.fields
if (isRestQuery(query, fields)) {
 // fields is now type checked to 'RestQueryFields
}
```